add remote debug option for otel-collector to builder (#6148)
open-telemetry/opentelemetry-collector: cmd/builder/README.md

Use `ocb --help` to learn about which flags are available.

## Debug

> To enable debug for otel-collector-binary set configuration flag `debug-compilation` to true.

Reviewer: ```suggestion
To keep the debug symbols in the resulting OpenTelemetry Collector binary, set the configuration property `debug_compilation` to true.
```

> Then install `go-delve` and run otel-collector with `dlv` command as following example:

Reviewer: ```suggestion
Then install `go-delve` and run OpenTelemetry Collector with `dlv` command as the following example:
```

```bash
# go install github.com/go-delve/delve/cmd/dlv@latest
# ~/go/bin/dlv --listen=:2345 --headless=true --api-version=2 --accept-multiclient --log exec .otel-collector-binary -- --config otel-collector-config.yaml
```

> Finally load otel-collector as a project into IDE, configure debug for Go

Reviewer: ```suggestion
Finally, load the OpenTelemetry Collector as a project in the IDE, configure debug for Go
```

Side note: how to do that? Do you have a link to an external document on how to "load as a project in the IDE"?

    output_path: /tmp/otelcol-distributionNNN # the path to write the output (sources and binary). Optional.
    version: "1.0.0" # the version for your custom OpenTelemetry Collector. Optional.
    go: "/usr/bin/go" # which Go binary to use to compile the generated sources. Optional.

>     debug-compilation: true # this option enable debugging possibility with go-delve tool. Optional.

Reviewer: ```suggestion
    debug_compilation: false # enabling this causes the builder to keep the debug symbols in the resulting binary. Optional.
```

Where possible, we show the default value here.
